docs/csharp/fundamentals/exceptions/creating-and-throwing-exceptions.md

```diff
@@ -44,15 +44,23 @@ The following list identifies practices to avoid when throwing exceptions:
 - Don't throw <xref:System.Exception?displayProperty=nameWithType>, <xref:System.SystemException?displayProperty=nameWithType>, <xref:System.NullReferenceException?displayProperty=nameWithType>, or <xref:System.IndexOutOfRangeException?displayProperty=nameWithType> intentionally from your own source code.
 - Don't create exceptions that can be thrown in debug mode but not release mode. To identify run-time errors during the development phase, use Debug Assert instead.
 
-## Defining Exception Classes
+## Exceptions in task-returning methods
+
+Methods declared with the `async` modifier have some special considerations when it comes to exceptions. Exceptions thrown in an `async` method are stored in the returned task and don't emerge until, for example, the task is awaited. For more information about stored exceptions, see [Asynchronous exceptions](../../asynchronous-programming/index.md#asynchronous-exceptions).
+
+We recommend that you validate arguments and throw any corresponding arguments, such as <xref:System.ArgumentException>, <xref:System.ArgumentNullException>, and <xref:System.ObjectDisposedException>, before entering the asynchronous parts of your methods. That is, these validation exceptions should emerge synchronously before the work starts. The following code snippet shows an example where, if the exceptions are thrown, the <xref:System.ArgumentException> exceptions would emerge synchronously, whereas the <xref:System.InvalidOperationException> would be stored in the returned task.
```

Comment:
FWIW, I'm a little surprised ObjectDisposedException is in this list.

Reply:
I added it because of this line: https://github.com/dotnet/runtime/blob/eaa8ea19f4b2f5e222f92707e8f7b1fc3bdd0da9/src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs#L563

But I also see a typo that I'll need to fix. Do you want me to remove it?